lua/entities/prop_vehicle_crawler/init.lua

```diff
@@ -0,0 +1,304 @@
+AddCSLuaFile("cl_init.lua")
+AddCSLuaFile("shared.lua")
+include("shared.lua")
+
+local CVAR_FASTDL = CreateConVar("prop_vehicle_crawler_fastdl", "1", FCVAR_ARCHIVE, "Should clients download content for crawlers on join or not")
+local Ride_Height = 33
+local tag = "prop_vehicle_crawler"
+
+--FastDL
+local function add_resource_dir(dir)
+    for _, f in pairs(file.Find(dir .. "/*","GAME")) do
+        local path = dir .. "/" .. f
+        resource.AddSingleFile(path)
+    end
+end
+
+if CVAR_FASTDL:GetBool() then
+    resource.AddSingleFile("materials/entities/prop_vehicle_crawler.png")
+
+    add_resource_dir("sound/crawler")
+    add_resource_dir("materials/models/crawler")
+    add_resource_dir("models/crawler")
+end
+
+function ENT:SetupTrails()
+    self.Trails = self.Trails or {}
+    for i = 1, 4 do
+        if IsValid(self.Trails[i]) then SafeRemoveEntity(self.Trails[i]) end
+
+        self.Trails[i] = util.SpriteTrail(self, i, self.EnergyColor, true, 32, 0, 0.1, 0.015625, "trails/laser.vmt")
+        if CPPI then
```

Comment:
what does CPPI come from ?
Also SetOwner is in vanilla gmod so you want to call that regardless of CPPI